src/test/java/StringManipulationTest.java

```diff
@@ -121,28 +133,36 @@ public void testRestoreString1()
     @Test
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Tes2t");
+        int [] array;
+        array=new int[]{0,1,2,4,3};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "Test2");
 
     }
 
     @Test
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
+        assertFalse(manipulatedstring.toString() == "", "String is null");
```

Comment:
You are not calling .restoreString here. If this test is to test the behavior of restoreString it should be getting called in this testCondition.

Reply:
adressed this too